irohad/model/execution/impl/command_executor.cpp

```diff
@@ -117,33 +122,35 @@ namespace iroha {
 
     bool CreateRoleExecutor::execute(const Command &command,
                                      ametsuchi::WsvQuery &queries,
-                                     ametsuchi::WsvCommand &commands) {
+                                     ametsuchi::WsvCommand &commands,
+                                     const std::string &creator_account_id) {
       auto cmd_value = static_cast<const CreateRole &>(command);
 
       return commands.insertRole(cmd_value.role_name)
           and commands.insertRolePermissions(cmd_value.role_name,
                                              cmd_value.permissions);
     }
 
-    bool CreateRoleExecutor::hasPermissions(const Command &command,
-                                            ametsuchi::WsvQuery &queries,
-                                            const Account &creator) {
-      creator_ = creator;
+    bool CreateRoleExecutor::hasPermissions(
+        const Command &command,
+        ametsuchi::WsvQuery &queries,
+        const std::string &creator_account_id) {
       return checkAccountRolePermission(
-          creator.account_id, queries, can_create_role);
+          creator_account_id, queries, can_create_role);
     }
 
     bool CreateRoleExecutor::isValid(const Command &command,
-                                     ametsuchi::WsvQuery &queries) {
+                                     ametsuchi::WsvQuery &queries,
+                                     const std::string &creator_account_id) {
       auto cmd_value = static_cast<const CreateRole &>(command);
       cmd_value.role_name.size();
 
       auto role_is_a_subset =
           std::all_of(cmd_value.permissions.begin(),
                       cmd_value.permissions.end(),
-                      [this, &queries](auto perm) {
+                      [&queries, &creator_account_id](auto perm) {
```

Comment:
`std::bind(checkAccountRolePermission, std::cref(creator_account_id), std::cref(queries), _1)` should do the thing